Cargo.toml

```diff
@@ -37,6 +37,7 @@ tree_magic = "0.2.1"
 snafu = "0.6.2"
 bytes = "0.5.3"
 hex = "0.3.1"
+futures = "*"
```

Comment:
Please use a specific version if possible, or at least restrict it to a minor version. You really don't want to be allowing any version of `futures` to be used here.

Reply:
Yes alright